### ISSUE TITLE
Hlf-ca not deploying correctly in AKS

### ISSUE THREAD

**Issue by CodingSinceThe80s** [aidtechnology/lf-k8s-hlf-webinar#3]
Hi guys, thanks again for posting this. I am following the README file and have the following issue after running:
`helm install stable/hlf-ca -n ca --namespace blockchain -f ./helm_values/ca_values.yaml`
I have edited the file to use my own hostname previous to running the above.
### Current state/repro steps:
Deployment works, but the ca is not coming up. Running 
`kubectl logs -n blockchain $CA_POD`
results in the following error:
```
2018/12/15 09:05:03 [DEBUG] CA Home Directory: /var/hyperledger/fabric-ca
2018/12/15 09:05:03 [DEBUG] Checking configuration file version '1.3.0' against server version: '1.2.0-stable'
2018/12/15 09:05:03 [DEBUG] Closing server DBs
Error: Configuration file version '1.3.0' is higher than server version '1.2.0-stable'
```
Upon seeing this I have tried to connect to the pod (interactive shell), but this is also not possible as the Pod is in the following error state:
```
pod has unbound PersistentVolumeClaims 
Unable to mount volumes for pod "ca-hlf-ca-798c98c84f-258tl_blockchain(45e33956-0047-11e9-bfa6-8666818dfc0c)": timeout expired waiting for volumes to attach or mount for pod "blockchain"/"ca-hlf-ca-798c98c84f-258tl". list of unmounted volumes=[data]. list of unattached volumes=[data ca-config default-token-b9n8t] 
Back-off restarting failed container
```
### Desired state:
Pod is coming up as required and is in running state

Would appreciate some feedback on this. If it helps I have had a similar issue which was related to the storageClassName property being set to "standard", whereas in Azure Kubernetes Service (AKS) the default storageClassName is "default". However I wasnt able to find this in the ca deployment, I am too inexperienced in K8s :)

**Comment by alexvicegrab** [comment]
Hi @CodingSinceThe80s, thanks so much for flagging it. I have added a patch. Please destroy your previous deployment and associated PVCs and try again, and let me know if this has addressed the issue.

**Comment by alexvicegrab** [comment]
@CodingSinceThe80s, any luck?

**Comment by CodingSinceThe80s** [comment]
Hi and thanks for checking in. The issue was that the AKS standard setting for storage is to create virtual drives. However there is a limitation of 5 per node (I did try this with one, two, then three nodes). If another storage request comes in it is denied. I wanted to go through the process first with only one node so I switched the provisioner to kubernetes.io/azure-file and created a separate storage account and the problems went away. It also helps if the AKS cluster is provisioned with the extra setting:„storageProfile“:“StorageAccount“ in the ARM template.
So long story short: now its working :) 


Gesendet von Yahoo Mail für iPhone


Am Mittwoch, Februar 27, 2019, 6:26 PM schrieb Sasha (Alejandro Vicente Grabovetsky) <notifications@github.com>:


@CodingSinceThe80s, any luck?

**Comment by alexvicegrab** [comment]
Aha, yes, I have seen this being an issue alas. Yes, azure-file can help with the drives but is a bit slower than actual drives (especially the Premium, i.e. SSDs).

Thanks for getting back to me with the update, I'll close the issue!
